sw/host/opentitanlib/src/transport/ultradebug/gpio.rs

```diff
@@ -31,6 +32,15 @@ impl UltradebugGpio {
             device: ultradebug.mpsse(ftdi::Interface::B)?,
         })
     }
+
+    pub fn enumerate() -> HashMap<String, u32> {
+        let mut pins = HashMap::new();
+        pins.insert(String::from("UD_SPI_ZB"), Self::PIN_SPI_ZB);
+        pins.insert(String::from("UD_RESET_B"), Self::PIN_RESET_B);
+        pins.insert(String::from("UD_BOOTSTRAP"), Self::PIN_BOOTSTRAP);
+        pins.insert(String::from("UD_TGT_RESET"), Self::PIN_TGT_RESET);
+        pins
```

Comment:
This will create a new HashMap every time, but in the backend's mapping of names to numbers, this is static and immutable data.

What you probably want is to define the mapping as a const, but Rust doesn't really have a way of doing that for HashMaps, nor does it have a nice syntax for HashMap literals.

The next best thing is to use a combination of macros to help you.  In, particular, you probably want to use [`lazy_static!`](https://crates.io/crates/lazy_static) and [`collection!`](https://github.com/cfrantz/opentitan/blob/ottool-cw310-for-jes/sw/host/opentitanlib/src/util/mod.rs#L11) (wip) to write something like [this](https://github.com/cfrantz/opentitan/blob/ottool-cw310-for-jes/sw/host/opentitanlib/src/transport/ultradebug/gpio.rs#L82).